tests/trainer/test_tp.py

```diff
@@ -47,3 +49,80 @@ def test_tp_train(world_size: int):
     )
 
     trainer.fit()
+
+
+@pytest.mark.gpu
+@world_size(4)
+@pytest.mark.skipif(version.parse(torch.__version__) < version.parse('2.3'), reason='requires PyTorch 2.3+')
+@pytest.mark.filterwarnings(r'ignore:.*\(TP\) is experimental.*:FutureWarning')
+def test_tp_with_param_groups(world_size: int):
+    from torch.distributed.tensor.parallel import ColwiseParallel, RowwiseParallel
+
+    model = SimpleModel()
+    dataset = RandomClassificationDataset(size=8)
+    dataloader = DataLoader(dataset, batch_size=2, sampler=dist.get_sampler(dataset))
+    optimizer = torch.optim.SGD([{
+        'params': model.fc1.parameters(),
+        'lr': 0.1,
+    }, {
+        'params': model.fc2.parameters(),
+        'lr': 0.5,
+    }])
+
+    layer_plan = {
+        'fc1': ColwiseParallel(),
+        'fc2': RowwiseParallel(),
+    }
+
+    expected_error = 'Multiple optimizer groups are not supported with tensor parallelism.'
+
+    with pytest.raises(RuntimeError, match=expected_error):
+        _ = Trainer(
+            model=model,
+            optimizers=optimizer,
+            train_dataloader=dataloader,
+            parallelism_config={
+                'tp': {
+                    'layer_plan': layer_plan,
+                    'tensor_parallel_degree': 2,
+                },
+                'fsdp': {},
+            },
+            max_duration='3ba',
+        )
+    gc.collect()
+
+
+@pytest.mark.gpu
+@world_size(4)
+@pytest.mark.skipif(version.parse(torch.__version__) < version.parse('2.3'), reason='requires PyTorch 2.3+')
+@pytest.mark.filterwarnings(r'ignore:.*\(TP\) is experimental.*:FutureWarning')
+def test_tp_with_subset_of_params(world_size: int):
+    from torch.distributed.tensor.parallel import ColwiseParallel
+
+    model = SimpleModel()
+    dataset = RandomClassificationDataset(size=8)
+    dataloader = DataLoader(dataset, batch_size=2, sampler=dist.get_sampler(dataset))
+    optimizer = torch.optim.SGD(model.fc1.parameters(), lr=0.1)
+
+    layer_plan = {
+        'fc1': ColwiseParallel(),
+    }
+
+    expected_error = 'Passing in a subset of model parameters to the optimizer is not supported with tensor parallelism.'
+
+    with pytest.raises(ValueError, match=expected_error):
+        _ = Trainer(
+            model=model,
+            optimizers=optimizer,
+            train_dataloader=dataloader,
+            parallelism_config={
+                'tp': {
+                    'layer_plan': layer_plan,
+                    'tensor_parallel_degree': 2,
+                },
+                'fsdp': {},
+            },
+            max_duration='3ba',
+        )
+    gc.collect()
```

Comment:
why is this necessary?

Reply:
I was following my previous impl here: https://github.com/mosaicml/composer/blob/98a87f3845aa19d0dee53767c448533ac93287b6/tests/trainer/test_fsdp_param_groups.py#L51, but I'll try removing and see if the test pass in that case.